lib/modules/manager/npm/extract/index.ts

```diff
@@ -201,8 +202,8 @@ export async function extractPackageFile(
         dep.commitMessageTopic = 'Yarn';
         constraints.yarn = dep.currentValue;
         if (
-          dep.currentValue.startsWith('2') ||
-          dep.currentValue.startsWith('3')
+          semver.valid(dep.currentValue) &&
+          semver.major(dep.currentValue) > 1
```

Comment:
@rarkins we should think about how to support ranges. 🤔

Reply:
I'm pretty sure `semver.major("^3.0.0")` would output '3' right? Do you have an example where this wouldn't work? Perhaps it needs an other solution